doc/source/concepts/concepts.rst

```diff
@@ -446,3 +447,133 @@ Below, we see an example of how to access the :code:`status` attribute of the re
+
+:code:`:__init__()`: The :code:`__init__()` method is the constructor that takes in variables
+including :code:`apply_fn` which is the callable function to be applied in the backend environment and a set of :code:`args` and :code:`kwargs`.
+When a :code:`Deps` object is initialized, the :code:`apply_fn`, :code:`args`, and :code:`kwargs` are serialized into transportable objects.
+
+:code:`:apply()`: The :code:`:apply()` function returns a :code:`Tuple` containing serialized copies of :code:`apply_fn`, :code:`args` and :code:`kwargs`.
+
+The following kinds of dependencies are supported at the electron level:
+
+* DepsPip - used in managing pip dependencies in an electron.
+* DepsBash - used in managing bash dependencies in an electron.
+* DepsCall - used in managing functions that are called before or after executing an electron.
+
+~~~~~~~
+DepsPip
+~~~~~~~
+:code:`DepsPip` is the class that is responsible for managing the list of required PyPI packages which have to be installed in the backend environment prior to executing the electron.
+
+:code:`__init__()`: The :code:`__init__()` constructor of :code:`DepsPip` takes a list of required PyPI packages and/or a path to the file that contains the list of requirements. It overrides :code:`Deps`'s :code:`__init_()` by
+using :code:`apply_pip_deps` which is a callable that installs the required PyPI packages.
+
+Below is an example of using :code:`DepsPip` to specify a list of PyPI packages in an electron::
+
+    import covalent as ct
+    from covalent import DepsPip
+
+    @ct.electron(
+        deps_pip=DepsPip(packages=["numpy==0.23", "qiskit"]),
+    )
+    def task():
+    ...
+
+Alternatively, one can specify the path to a :code:`requirements.txt` file that contains the list of required packages::
+
+    @ct.electron(
+        deps_pip=DepsPip(reqs_path="/usr/foo/requirements.txt")
+    )
+
+~~~~~~~~
+DepsBash
+~~~~~~~~
+
+:code:`DepsBash` is the class that is responsible for managing the execution of bash commands that are required by an electron.
+
+:code:`__init_()`: The :code:`__init__()` constructor of :code:`DepsBash` accepts a list of bash commands as its argument.
+It overrides :code:`Deps`'s :code:`__init__()` by accepting :code:`apply_bash_commands` which is the callable that executes the commands and :code:`apply_args`
+which references the specified list of commands.
+
+:code:`apply_bash_commands`: This takes the list of commands and executes them as subprocesses in the same environment as the electron.
+
+Below is an example of using :code:`DepsBash` to specify a list of bash commands in an electron::
+
+    import covalent as ct
+    from covalent import DepsBash
+
+    @ct.electron(
+        deps_bash=DepsBash(["echo $PATH", "ssh foo@bar.com"]),
+    )
+    def task():
+    ...
+
+~~~~~~~~
+DepsCall
```

Comment:
@Emmanuel289 quick comment, apparently I did not know as well, but seems like `DepsCall` is the base class for all other Deps, which means one can pass on a list of any Deps function to `call_after` and `call_before` even bash commands and it will work.

Reply:
@santoshkumarradha Ohh I didn't know that too. Does that mean that the Deps from which DepsPip and DepsBash inherits, now takes a backseat to DepsCall as the super parent?

Reply:
Hi @santoshkumarradha, see https://github.com/AgnostiqHQ/covalent/pull/775/commits/7f0c46347c4158c91341c04cb0a08a8477918fe9 related to DepsCall